frontend/src/scene/spinner/LoadingSpinner.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import styled from 'react-emotion';
+import Modal, { Content, Title } from '../../components/modal';
+const SmallModal = styled(Modal)`
+    height: 40%;
+    display: flex;
+    text-align: center;
+    div {
+        font-size: 2.5rem;
+    }
+`;
+const CustomTitle = styled(Title)`
+    font-weight: 400;
+
+    strong {
+        color: ${(props) => props.theme.main};
+    }
+`;
+
+const Spinner = styled(Content)`
+    div,
+    div:after {
+        border-radius: 50%;
+        width: 10em;
+        height: 10em;
+    }
+    div {
+        margin: 0rem auto !important;
+        font-size: 1.5rem;
+        position: relative;
+        padding: 0rem !important;
+        border-top: 1.1em solid #abb2b9;
+        border-right: 1.1em solid #abb2b9;
+        border-bottom: 1.1em solid #abb2b9;
+        border-left: 1.1em solid #ffffff;
+        -webkit-transform: translateZ(0);
+        -ms-transform: translateZ(0);
+        transform: translateZ(0);
+        -webkit-animation: load8 1.1s infinite linear;
+        animation: load8 1.1s infinite linear;
+    }
+    span {
+        margin: 0px !important;
+    }
+    @-webkit-keyframes load8 {
+        0% {
+            -webkit-transform: rotate(0deg);
+            transform: rotate(0deg);
+        }
+        100% {
+            -webkit-transform: rotate(360deg);
+            transform: rotate(360deg);
+        }
+    }
+    @keyframes load8 {
+        0% {
+            -webkit-transform: rotate(0deg);
+            transform: rotate(0deg);
+        }
+        100% {
+            -webkit-transform: rotate(360deg);
+            transform: rotate(360deg);
+        }
+    }
+`;
+
+export default class LoadingSpinner extends React.Component {
+    render() {
+        const { i18nContent, show, isPaymentRedirect } = this.props;
+        return (
+            <SmallModal show={show} hideCloseButton>
+                <CustomTitle>{i18nContent.loadingSpinner.title}</CustomTitle>
+                <div>
+                    {isPaymentRedirect ? (
```

Comment:
This "Loading spinner" knows too much of the codebase, it seemed

Reply:
yeah why dont you pass the string message from the component which knows it in the first place?

Reply:
updated.